lib/AST/ASTMangler.cpp

```diff
@@ -2356,29 +2356,32 @@ CanType ASTMangler::getDeclTypeForMangling(
   }
 
 
-  CanType type = decl->getInterfaceType()
-                      ->getReferenceStorageReferent()
-                      ->getCanonicalType();
-  if (auto gft = dyn_cast<GenericFunctionType>(type)) {
+  Type type = decl->getInterfaceType()
+                  ->getReferenceStorageReferent();
+  if (type->hasArchetype())
```

Comment:
Can you file a radar and assign it to me to fix the underlying problem here? I believe the only decls that are affected by this are ParamDecls (in fact, can you add an assert that if hasArchetype() is true, the decl is a ParamDecl?)

Reply:
Thanks Slava! I filed rdar://problem/55268049 and updated this PR with the assertion.